sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/WindowTest.java

```diff
@@ -693,46 +696,47 @@ public void verifyDeterministic() throws NonDeterministicException {
   }
 
   private static class CustomWindowFn<T> extends WindowFn<T, CustomWindow> {
-
-    @Override public Collection<CustomWindow> assignWindows(AssignContext c) throws Exception {
+    @Override
+    public Collection<CustomWindow> assignWindows(AssignContext c) throws Exception {
       String element;
       // It loses genericity of type T but this is not a big deal for a test.
       // And it allows to avoid duplicating CustomWindowFn to support PCollection<KV>
-      if (c.element() instanceof KV){
+      if (c.element() instanceof KV) {
         element = ((KV<Integer, String>) c.element()).getValue();
       } else {
         element = (String) c.element();
       }
       // put big elements in windows of 30s and small ones in windows of 5s
       if ("big".equals(element)) {
         return Collections.singletonList(
-            new CustomWindow(c.timestamp(), c.timestamp().plus(Duration.standardSeconds(30)),
-                true));
+            new CustomWindow(
+                c.timestamp(), c.timestamp().plus(Duration.standardSeconds(30)), true));
       } else {
         return Collections.singletonList(
-            new CustomWindow(c.timestamp(), c.timestamp().plus(Duration.standardSeconds(5)),
-                false));
+            new CustomWindow(
+                c.timestamp(), c.timestamp().plus(Duration.standardSeconds(5)), false));
       }
     }
 
     @Override
     public void mergeWindows(MergeContext c) throws Exception {
-      List<CustomWindow> toBeMerged = new ArrayList<>();
-      CustomWindow bigWindow = null;
-      for (CustomWindow customWindow : c.windows()) {
-        if (customWindow.isBig) {
-          bigWindow = customWindow;
-          toBeMerged.add(customWindow);
-        } else if (bigWindow != null
-            && customWindow.start().isAfter(bigWindow.start())
-            && customWindow.end().isBefore(bigWindow.end())) {
-          toBeMerged.add(customWindow);
+      Map<CustomWindow, Set<CustomWindow>> windowsToMerge = new HashMap<>();
```

Comment:
Yes, that is what I had understood. Thanks for pointing out this bug and fixing it !